docs/sql-guide/functions/function-coalesce.md

```diff
@@ -0,0 +1,116 @@
+---
+title: COALESCE()
+layout: default
+parent: Functions
+grand_parent: SQL guide
+---
+
+# COALESCE() function
+
+`COALESCE()` function is used to return the first non-null value among the provided parameters.
+
+## Syntax
+
+```
+coalesce(value1, value2, ...)
+```
+
+## Arguments
+
+| Argument | Description | Length |
+|---|---|---|---|
+| `value1`,`value2`.. | Any expressions or values of same data type. The function returns the first non-null value among them. | 1 |
+
+## Returns
+
+The data type and value of the first non-null parameter
+
+## Examples
+
+### using Coalesce() function to handle null values
+
+```sql
+create table stock
+    (__id id, product string, brand string, subcategory string, category string, family string, quantity_available int, minimum_to_have int);
+
+insert into stock
+    values 
+    (1, 'pork ribs', NULL, 'pork meat', 'meat','food',400, 130),
+    (2, 'tomatoes','Mr Red', NULL, 'vegetables','food',280, 100),
+    (3,'lettuce',NULL, 'Leaf vegetables', NULL,'food',280, Null),
+    (4,'Bananas',NULL, NULL,NULL,'food',Null, NULL),
+    (5,'hamburger','MaxBurg','cow meat','meat','food',220, 150),
+    (6,'hamburger','SuperBurga',Null,Null,Null,125, Null);
+
+select * from stock;
+
++----+----------+----------+----------------+-----------+--------+-------------------+----------------+
+|_id | product  | brand    | subcategory    | category  | family | quantity_available| minimum_to_have|
++----+----------+----------+----------------+-----------+--------+-------------------+----------------+
+| 1  | pork ribs| NULL     | pork meat      | meat      | food   | 400               | 130            |
++----+----------+----------+----------------+-----------+--------+-------------------+----------------+
+| 2  | tomatoes | Mr Red   | NULL           | vegetables| food   | 280               | 100            |
++----+----------+----------+----------------+-----------+--------+-------------------+----------------+
+| 3  | lettuce  | NULL     | Leaf vegetables| NULL      | food   | 280               | NULL           |
++----+----------+----------+----------------+-----------+--------+-------------------+----------------+
+| 4  | Bananas  | NULL     | NULL           | NULL      | food   | NULL              | NULL           |
++----+----------+----------+----------------+-----------+--------+-------------------+----------------+
+| 5  | hamburger| MaxBurg  | cow meat       | meat      | food   | 220               | 150            |
++----+----------+----------+----------------+-----------+--------+-------------------+----------------+
+| 6  | hamburger|SuperBurga| NULL           | NULL      | NULL   | 125               | NULL           |
++----+----------+----------+----------------+-----------+--------+-------------------+----------------+
+
+select product,coalesce(brand, 'locally grown') as final_brand from stock;
```

Comment:
Where does 'locally grown' come from? It doesn't appear to be in the table?

Reply:
Is it a substitution for a null value? If so, that needs to go into the syntax and arguments, OR under an `## Additional information` heading.

e.g.,

```
## Additional information

### Null substitutions in results

Null values can be substituted for a quoted string in results.
```
Also, if this is true, could `coalesce()` be used like a find&replace in an INSERT statement?

Reply:
it can be any value of type as the value we are trying to replace. It can be int, decimal, string or anything..

Reply:
its not find and replace, it will only replace the null values.. non-null values would be returned as is. Its part of Select statement not Insert. It kind of creates a view which has the replaced value.. we are not changing anything in the actual table

Reply:
My query is around the syntax.
Based on your example,  `COALESCE( value1, value2...)` is actually `COALESCE (colname, <value-substitution>)`
So my question is this:
1. Is the column name required for all COALESCE() queries?
2. If not, what else could you substitute for `value1, value2`? (e.g., could a user do a nested select statement, or a nested function)? If so this needs to be documented.

I

Reply:
fyi, here is what the source table looks like if that would provide some clarity
 select * from stock;
 _id | product   | brand      | subcategory     | category   | family | quantity_available | minimum_to_have
-----|------------|-----------|-----------------|------------|--------|--------------------|-----------------
   |1 | pork ribs | NULL       | pork meat       | meat       | food   | 400                | 130|
   |2 | tomatoes  | Mr Red     | NULL            | vegetables | food   | NULL               | 100|
   |3 | lettuce   | NULL       | Leaf vegetables | NULL       | food   | 280                | NULL|
   |4 | Bananas   | NULL       | NULL            | NULL       | food   | NULL               | NULL|
   |5 | hamburger | MaxBurg    | cow meat        | meat       | food   | 220                | 45|
   |6 | hamburger | SuperBurga | NULL            | NULL       | NULL   | 400                | 130|

Reply:
Yeah, totally.
So we're saying `COALESCE({<column-name>,...} {,'<null-replacement-value>'}>` is the syntax
One or more column names that replace null values with the quoted string